packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkInfo.spec.js

```diff
@@ -0,0 +1,259 @@
+    const errorSplit = message.split('\n')
+    throw errorSplit[0]
+  } else throw message instanceof Error ? message : new Error(message)
+}
+
+function getShallowWrapper(
+  link = {
+    token: '122235445488',
+    url: 'some-link'
+  }
+) {
+  return shallowMount(LinkInfo, {
+    localVue,
+    stubs: { ...stubs, 'oc-tag': true },
+    propsData: {
+      link: link
+    },
+    directives: {
+      'oc-tooltip': jest.fn()
+    }
+  })
+}
+
+function getMountedWrapper(
+  link = {
+    token: '122235445488',
+    url: 'some-link'
+  }
+) {
+  localVue.use(DesignSystem)
+  return mount(LinkInfo, {
+    localVue,
+    propsData: {
+      link: link
+    },
+    stubs,
+    directives: {
+      'oc-tooltip': jest.fn()
+    }
+  })
+}
+
+describe('LinkInfo', () => {
+  describe('link name', () => {
+    it('should show token as link name if link does not have name', () => {
+      const wrapper = getShallowWrapper()
+
+      expect(wrapper.vm.linkName).toEqual('122235445488')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('122235445488')
+    })
+
+    it('should show link name if provided link has name attribute', () => {
+      const wrapper = getShallowWrapper({
+        name: 'some-name',
+        url: 'some-link'
+      })
+
+      expect(wrapper.vm.linkName).toEqual('some-name')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('some-name')
+    })
+
+    it('should show link name if provided link has both name and token attribute', () => {
+      const wrapper = getShallowWrapper({
+        name: 'some-name',
+        token: 'some-token',
+        url: 'some-link'
+      })
+
+      expect(wrapper.vm.linkName).toEqual('some-name')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('some-name')
+    })
+  })
+
+  describe('link prop', () => {
+    it('should be required', () => {
+      expect(() => {
+        shallowMount(LinkInfo, {
+          localVue,
+          stubs: { ...stubs, 'oc-tag': true }
+        })
+      }).toThrow('[Vue warn]: Missing required prop: "link"')
+    })
+
+    it('should not accept values other than type object', () => {
+      expect(() => {
+        shallowMount(LinkInfo, {
+          localVue,
+          stubs: { ...stubs, 'oc-tag': true },
+          propsData: {
+            link: 'some-link'
+          }
+        })
+      }).toThrow(
+        '[Vue warn]: Invalid prop: type check failed for prop "link". Expected Object, got String with value "some-link".'
+      )
+    })
+  })
+  describe('link url', () => {
+    it('should set provided link url as href attribute of link tag and as value of copy to clipboard button', () => {
+      const wrapper = getShallowWrapper({
+        url: 'some-link'
+      })
+
+      expect(wrapper.vm.link.url).toEqual('some-link')
+      expect(wrapper.find('a').attributes().href).toEqual('some-link')
+      expect(wrapper.find('.oc-files-public-link-copy-url').attributes().value).toEqual('some-link')
```

Comment:
put the selectors in a seperate object and also use a bit more specific selector for the link

Reply:
fixed